services/parser-database/apiSpec.yaml

```diff
@@ -0,0 +1,134 @@
+  version: "1.0.0"
+  title: "Parser/Database Component"
+  contact:
+    email: "jaimehisao@google.com"
+host: "localhost"
+basePath: "/v1"
+schemes:
+- "https"
+- "http"
+paths:
+  /parse:
+    post:
+      tags:
+      - "parser"
+      summary: "Trigger a document download, parse, and metadata tagging"
+      responses:
+        "200":
+         description: "Document was parsed successfully"
+        "409":
+          description: "Parser service is already running."
+        "500":
+          description: "Server cannot parse the document and add it to the database. Likely if the document is not in the specified URL or if the database is offline."
+  /articles/byKeyword:
+    post:
+      tags:
+      - "articles"
+      summary: "Finds any article that contains the specified keyword(s)"
+      description: ""
+      operationId: "returnArticleIDs"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "Keywords that need to be matched with an article in the database"
+        required: true
+        schema:
+          $ref: '#/definitions/ArticleKeywordsQuery'
+      responses:
+        "200":
+          description: "Successful Operation"
+          schema:
+            $ref: '#/definitions/ArticleKeywords'
+  /articles/{id}:
+    get:
+      tags:
+      - "articles"
+      summary: "Returns the text of the requested article"
+      description: "Single article ID value"
+      operationId: "findTextByID"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "id"
+        in: "path"
+        required: true
+        description: "ID of the requested article"
+        type: "string"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Article"
+        "404":
+          description: "No article found with that ID"
+        "500":
+          description: "Error when retrieving the article"
+definitions:
+  Article:
+    type: "object"
+    properties:
+      articleNumber:
+        type: "integer"
+        format: "int 64"
+      text:
+        type: "string"
+      wordCount:
+        type: "integer"
+        format: "int 64"
+    required: 
+    - articleNumber
+    - text
+    - wordCount
+  ArticleKeywordsQuery:
+    type: "object"
+    properties:
+       keywords:
+        type: "array"
+        items:
+          type: "string"
+    required:
+    - keywords
+  ArticleKeywords:
+    type: "object"
+    properties:
+      keyword:
+        type: "array"
+        items:
+          type: "object"
+          $ref: '#/definitions/ArticleKeywordFrequencyPair'
+    required: 
+    - keyword
+  # For Reference Only
+  ArticleKeywordFrequencyPair:
+    type: "object"
+    properties:
+      articleId:
+        type: "string"
+      frequency:
+        type: "integer"
+    required:
+    - articleId
+    - frequency
```

Comment:
This response object needs to include the keywords as well, right? Otherwise there is no way to figure out which keyword from the input is being referenced since the input is a list of keywords.